File: SAMM3.0/testArea.m
beam1 = Beam(Electron);
beam1.energy = 4.5*PhysicalUnits.MeV;
ptcle1 = [0.001,0,0,0,0,0]';
ptcle2 = [0,0,0.001,0,0,0]';
ptcle3 = [0,0,0,0,0,0]';
beam1.particles = [ptcle1, ptcle2, ptcle3];

drift = Drift;
drift.length = 0.1;
drift.name = 'drift1';
quad1 = Quadrupole;
quad1.length = 0.05;
quad1.name = 'quad1';
quad1.gradient = 0.00;
dip1 = Dipole;
dip1.length = 0.4;
dip1.name = 'dip1';
dip1.angle = pi/4;
dip1.field = beam1.rigidity * dip1.angle / dip1.length;


beamlineD = Beamline;
beamlineD.componentlist = {,dip1};
beam2 = beamlineD.Track([1,1],beam1);
beam2 = beam2.particles

%[G,r] = ComputeTransferMatrix(beamlineD, [1 2], beam1, [0 0 0 0 0 0]');

File: SAMPL/testing/testArea.py
import os
import sys
sys.path.append(str(os.path.dirname(os.path.abspath(__file__)))+'\\..\\')
sys.path.append(str(os.path.dirname(os.path.abspath(__file__)))+'\\..\\..\\')
from SAMPL.sourceCode.SAMPLcore.Components import Drift as d
from SAMPL.sourceCode.SAMPLcore.Components import Dipole as D
from SAMPL.sourceCode.SAMPLcore.Components import Quadrupole as Q
from SAMPL.sourceCode.SAMPLcore.Components import Screen as S
from SAMPL.sourceCode.SAMPLcore.Components import OrbitCorrector as C
from SAMPL.sourceCode.SAMPLcore.Components import BeamPositionMonitor as BPM
from SAMPL.sourceCode.SAMPLcore.SAMPLlab import Beamline
from SAMPL.sourceCode.SAMPLcore.SAMPLlab import Beam
from SAMPL.sourceCode.SAMPLcore.Particles import Electron
from SAMPL.sourceCode.SAMPLcore.SAMPLlab import PhysicalUnits
import SAMPL.sourceCode.SAMPLcore.Physics.ComputeTransferMatrix as CTM
import numpy as np


beam1 = Beam.Beam(species=Electron.Electron, energy=4.5 * PhysicalUnits.MeV)
ptcle1 = [0.001, 0, 0, 0, 0, 0]
ptcle2 = [0, 0, 0.001, 0, 0, 0]
ptcle3 = [0, 0, 0, 0, 0, 0]
beam1.particles = np.array([ptcle1, ptcle2, ptcle3])

Line = Beamline.Beamline(componentlist=[])
Line.componentlist.append(d.Drift(name='drift1', length=0.1))
Line.componentlist.append(Q.Quadrupole(name='quad1', gradient=0.0, length=0.05))

beam2 = Line.TrackMatlab([0, len(Line.componentlist) - 1], beam1)

traj = np.array([0, 0, 0, 0, 0, 0])
a, b = CTM.ComputeTransferMatrix(Line, [0, 1], beam1, traj)

print b
print a
